app/assets/javascripts/darkswarm/directives/enterprise_modal.js.coffee

```diff
@@ -1,9 +1,15 @@
-Darkswarm.directive "enterpriseModal", ($modal)->
+Darkswarm.directive "enterpriseModal", ($modal, Enterprises, EnterpriseResource) ->
   restrict: 'E'
   replace: true
   template: "<a ng-transclude></a>"
   transclude: true
-  link: (scope, elem, attrs, ctrl)->
-    elem.on "click", (ev)=>
+  link: (scope, elem, attrs, ctrl) ->
+    elem.on "click", (ev) =>
       ev.stopPropagation()
+      params =
+        id: scope.enterprise.id
+      EnterpriseResource.relatives params, (data) =>
+        Enterprises.addEnterprises data
+        scope.enterprise = Enterprises.enterprises_by_id[scope.enterprise.id]
+        Enterprises.dereferenceEnterprise scope.enterprise
```

Comment:
It could, but it's not helpful in this case.
When a user opens the modal of a producer, the relatives of that producer are loaded and added. These enterprises don't need dereferencing in this case. The producer needs dereferencing, so that its shops can be displayed.